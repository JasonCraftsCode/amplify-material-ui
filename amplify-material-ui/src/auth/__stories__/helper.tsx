import * as React from 'react';
import { AuthContext } from '../auth-context';

export const withContext = ({
    authState,
    onStateChange = () => {},
}) => component => (
    <AuthContext.Provider value={{ authState, onStateChange }}>
        {component}
    </AuthContext.Provider>
);
