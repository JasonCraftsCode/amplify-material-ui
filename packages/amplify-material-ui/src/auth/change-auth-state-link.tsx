import * as React from 'react';
import { Link } from '@material-ui/core';

import { useAuthContext } from './auth-context';

export interface ChangeAuthStateLinkProps {
    label: string;
    newState: string;
    authData?: any;
}

export const ChangeAuthStateLink: React.FC<ChangeAuthStateLinkProps> = props => {
    const { label, newState, authData } = props;

    const { onStateChange } = useAuthContext();

    return (
        <Link
            href="#"
            onClick={(): void => onStateChange(newState, authData)}
            variant="body2"
        >
            {label}
        </Link>
    );
};
