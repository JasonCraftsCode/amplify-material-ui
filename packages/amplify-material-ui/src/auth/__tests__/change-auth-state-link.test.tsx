import * as React from 'react';
import { render } from '@testing-library/react';

import { withContext } from './helper';
import { ChangeAuthStateLink } from '../change-auth-state-link';

describe('change-auth-state-link', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(
            withContext(
                <ChangeAuthStateLink label="LinkLabel" newState="newState" />,
            )(),
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
