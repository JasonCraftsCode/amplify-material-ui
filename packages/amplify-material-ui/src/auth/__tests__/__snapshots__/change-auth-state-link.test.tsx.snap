// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change-auth-state-link should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
    href="#"
  >
    LinkLabel
  </a>
</DocumentFragment>
`;
